### DISCUSSION TITLE
[warning] an anonymous plugin or theme was detected and ignored

### DISCUSSION THREAD

**Discussion by mrdrivingduck** [vuepress/core#878] [Q&A] [answered]
Hi~

I'm upgrading from `2.0.0-beta.35` to `2.0.0-beta.43`. I got warnings like:

```bash
$ node -v         
v17.4.0

$ npm -v          
8.5.0

$ npm run docs:dev

> polardb_pg_documentation@0.0.1 docs:dev
> vuepress dev docs

warning an anonymous plugin or theme was detected and ignored
warning an anonymous plugin or theme was detected and ignored
info Initializing VuePress and preparing data...

  vite v2.9.8 dev server running at:

  > Local:    http://localhost:8080/PolarDB-for-PostgreSQL/
  > Network:  http://30.225.184.37:8080/PolarDB-for-PostgreSQL/
下午3:47:08 [vite] ✨ optimized dependencies changed. reloading
```

And my web page cannot display as expected, the content on navbar has gone, and the logo on the navbar disappears, either. The theme goes dark even if I have turned off the dark mode. The browser console does not show any error or warnings.

Have I mis-config anything?

**Comment by Mister-Hope** [comment, marked as answer]
Check whether all your theme and plugins are using a `name` option and set them as package name. If not, submit an issue in their repo.

See https://v2.vuepress.vuejs.org/reference/plugin-api.html#name

**Reply by mrdrivingduck** [reply]
I am using the default theme of VuePress, and the official Algolia @vuepress/plugin-docsearch plugin.

```json
"devDependencies": {
    "@vuepress/plugin-docsearch": "^2.0.0-beta.43",
    "prettier": "2.6.2",
    "vuepress": "^2.0.0-beta.43"
}
```

Aren't they up-to-date together?

**Reply by Mister-Hope** [reply]
Check your config file please

**Reply by mrdrivingduck** [reply]
ok. I rewrite my config from JS to TS, thanks to VuePress's docs as template. Now the problem disappears. I'm not familiar with TS at all, never mind. 😅

**Comment by darekkay** [comment]
I had encountered the same issue with a local plugin. For other people debugging this issue, those two changes were required:

1. Adjust the plugin initialization

Before:

```js
plugins: [
    [
      require("./plugins/rss"),
      options
    ],
]
```

After:

```js
const rssPlugin = require("./plugins/rss");
...
plugins: [
    rssPlugin(options),
  ],
```

2. Move `ctx` in the plugin itself

Before:

```js
module.exports = (pluginOptions, ctx) => {
  onGenerated() {
    const pages = ctx.pages;
    ...
```

After:

```js
module.exports = (pluginOptions) => {
  onGenerated(app) {
    const pages = app.pages;
    ...
```

See this commit for everything I had to do to migrate from `beta.27` to `beta.45`:
https://github.com/darekkay/dashboard/commit/e2c02ffddda72638f1e32a466b21349e63c2050d

**Comment by seepine** [comment]
how fix it ?

**Comment by seepine** [comment]
`2.0.0-beta.35` is running
but upgrade to `2.0.0-beta.49`  it show me , and can not work
```
warning an anonymous plugin or theme was detected and ignored
```

the option 
```
plugins: [
    [
      "vuepress-plugin-demo-block-vue3",
      {
        componentsDir: path.resolve(__dirname, "./../../examples"),
      },
    ],
  ],
```
